plugins/blip/basics/chartbasicsfactory.js

```diff
@@ -42,17 +43,16 @@ var BasicsChart = React.createClass({
     bgClasses: React.PropTypes.object.isRequired,
     bgUnits: React.PropTypes.string.isRequired,
     onSelectDay: React.PropTypes.func.isRequired,
+    patient: React.PropTypes.object.isRequired,
     patientData: React.PropTypes.object.isRequired,
     timePrefs: React.PropTypes.object.isRequired,
     updateBasicsData: React.PropTypes.func.isRequired,
+    updateBasicsSettings: React.PropTypes.func.isRequired,
     trackMetric: React.PropTypes.func.isRequired,
   },
   _adjustSectionsBasedOnAvailableData: function(basicsData) {
-    if (_.isEmpty(basicsData.data.reservoirChange.data)) {
-      var siteChangeSection = _.find(basicsData.sections, function(section) {
-        return section.type === 'reservoirChange';
-      });
-      siteChangeSection.active = false;
+    if (basicsData.sections.siteChanges.type !== constants.SECTION_TYPE_UNDECLARED && _.isEmpty(basicsData.data[basicsData.sections.siteChanges.type].data)) {
```

Comment:
I don't think this logic is _quite_ right... Because we have to look back to the time prior to the start of the time period covered by the Basics to calculate the length of the first site change, we put all of the prime & reservoirChange events in the basics data, but if they are all > 3 weeks ago, then the array will have content but the relevant data is still missing.

To test my instinct that this wasn't quite the right logic, I went into my local Mongo and changed my `patient.settings.siteChangeSource` to `reservoirChange` because I have lots of OmniPod data in my account, but none of it is recent. Now the app is crashing entirely because it's actually looking for an `infusionSiteHistory` under the reservoirChange data but that doesn't exist because of the age of the relevant data.

So both the logic here and the crash issue should be addressed. I think you'll be able to work on it yourself locally if you just have some old OmniPod files to upload locally alongside your Tandem - I'll e-mail you some .ibfs, but let me know if you can't replicate the problem.